maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/elasticsearch/SearchAdapterHelper.java

```diff
@@ -211,7 +211,8 @@ public static Script getInline(Map<String, Object> parameters) {
             "if (!ctx._source.repositories.contains(params.repository)) { ctx._source.repositories.add(params.repository) } \n"
                 + "ctx._source.analysis_state = params.analysis_state;\n"
                 + "ctx._source.updated_at = ZonedDateTime.parse(params.updated_at).toInstant().toEpochMilli();\n"
-                + "if (params.published_at != null) { ctx._source.published_at = ZonedDateTime.parse(params.published_at).toInstant().toEpochMilli(); }\n",
+                + "if (params.published_at != null) { ctx._source.published_at = ZonedDateTime.parse(params.published_at).toInstant().toEpochMilli(); }\n"
+                + "if (params.first_published_at != null) { ctx._source.first_published_at = ZonedDateTime.parse(params.first_published_at).toInstant().toEpochMilli(); }\n",
```

Comment:
is there a reason to use this upsert script ? because this field cannot be updated I don't think it should be included here

Reply:
answered on slack, this field is null at first then gets value on first publish so it needs to be updated